libopae/src/sysfs.c

```diff
@@ -632,18 +626,22 @@ fpga_result sysfs_bdf_from_path(const char *sysfspath, int *b, int *d, int *f)
 
 	// 0123456
 	// bb:dd.f
-	*f = (int) strtoul(p+6, NULL, 16);
+	*f = (int)strtoul(p + 6, NULL, 16);
 	*(p + 5) = 0;
 
-	*d = (int) strtoul(p+3, NULL, 16);
+	*d = (int)strtoul(p + 3, NULL, 16);
 	*(p + 2) = 0;
 
-	*b = (int) strtoul(p, NULL, 16);
+	*b = (int)strtoul(p, NULL, 16);
+	*(p - 1) = 0;
```

Comment:
Can we add a check that `p` actually points far enough into the buffer that we don't walk into previous values on the stack (same for line 638 below)?

Reply:
I *can* add a check, but the "p += 6" above pretty much ensures it'll be OK, since the strrchr() found a '/' at 'p'.